tests/numerics/test_block_diag_solver.py

```diff
@@ -0,0 +1,117 @@
+import numpy as np
+import porepy as pp
+import pytest
+import scipy.sparse as sps
+
+import pygeon as pg
+
+
+@pytest.fixture()
+def M_sparse():
```

Comment:
don't you get complains from mypy about missing return types? also in most of the functions used for the tests